lib/userdb.c

```diff
@@ -35,8 +35,11 @@ resolve_user(const char *user, gint *uid)
   struct passwd *pw;
   gchar *endptr;
 
+  if (!uid)
+    return FALSE;
   *uid = 0;
-  if (!(*user))
+
+  if (!user || !(*user))
```

Comment:
The !user part should be assert instead. Return for !(*user) with false is fine.